### ISSUE TITLE
what does "balanced" option do?

### ISSUE THREAD

**Issue by erogol** [karandwivedi42/imagenet-multiGPU.torchnet#1]
I trace the code but don't understand what below code line and so that option. Could you explain ?

https://github.com/karandwivedi42/imagenet-multiGPU.torchnet/blob/master/main.lua#L179

**Comment by karandwivedi42** [comment]
Setting balanced to true does the training set sampling in class balanced manner.
https://github.com/karandwivedi42/imagenet-multiGPU.torchnet/blob/master/randomdataset.lua#L32

Just like how ShuffleDataset can be used with full size of underlying dataset or a different size, the RandomDataset can also be used like that.
